test/src/test/java/hudson/model/QueueTest.java

```diff
@@ -780,4 +782,24 @@ public void run() {
                 "B finished at " + buildBEndTime + ", C started at " + buildC.getStartTimeInMillis(), 
                 buildC.getStartTimeInMillis() >= buildBEndTime);
     }
+
+    @Issue("JENKINS-30084")
+    @Test
+    /*
+     * When a flyweight task is restricted to run on a specific node, the node will be provisioned
+     * and the flyweight task will be executed.
+     */
+    public void shouldRunFlyweightTaskOnProvisionedNodeWhenNodeRestricted() throws Exception {
+        MatrixProject project = r.createMatrixProject();
+        project.setAxes(new AxisList(
+                new Axis("axis", "a", "b")
+        ));
+        Label label = LabelExpression.get("aws-linux-dummy");
+        DummyCloudImpl dummyCloud = new DummyCloudImpl(r, 0);
+        dummyCloud.label = label;
+        r.jenkins.clouds.add(dummyCloud);
+        project.setAssignedLabel(label);
+        r.assertBuildStatusSuccess(project.scheduleBuild2(0));
```

Comment:
After the build ti makes sense to ensure the FlyWeight task has been launched on the cloud